src/components/connect.tsx

```diff
@@ -543,11 +542,6 @@ function connect<
       areMergedPropsEqual,
     }
 
-    // If we aren't running in "pure" mode, we don't want to memoize values.
-    // To avoid conditionally calling hooks, we fall back to a tiny wrapper
-    // that just executes the given callback immediately.
-    const usePureOnlyMemo = pure ? useMemo : (callback: () => any) => callback()
```

Comment:
Note that this actually changes behavior but ive assumed that to the more preferred one. The alternative is to inline the memoized value - as thay was effectively happening before my changes